internal/resource/demo/demo.go

```diff
@@ -84,6 +85,32 @@ func Register(r resource.Registry) {
 		}
 	}
 
+	validateArtistFn := func(res *pbresource.Resource) error {
+		if proto.Equal(res.Id.Type, TypeV1Artist) {
+			artist := &pbdemov1.Artist{}
+			if err := anypb.UnmarshalTo(res.Data, artist, proto.UnmarshalOptions{}); err != nil {
+				return err
+			}
+			if artist.Name == "" {
+				return fmt.Errorf("artist.name required")
+			}
+			return nil
+		}
+
+		if proto.Equal(res.Id.Type, TypeV2Artist) {
+			artist := &pbdemov2.Artist{}
+			if err := anypb.UnmarshalTo(res.Data, artist, proto.UnmarshalOptions{}); err != nil {
+				return err
+			}
+			if artist.Name == "" {
+				return fmt.Errorf("artist.name required")
+			}
+			return nil
+		}
+
+		return fmt.Errorf("unsupported artist type %v", resource.ToGVK(res.Id.Type))
+	}
```

Comment:
Is there any reason not to split this into two separate functions?

Reply:
🤦 Fixed in https://github.com/hashicorp/consul/pull/16950/commits/f47554ba639d56e060827dec87b55d89201b6fa9